lib/eslint/eslint.js

```diff
@@ -312,7 +326,13 @@ async function loadFlatConfigFile(filePath) {
         delete require.cache[filePath];
     }
 
-    const config = (await import(fileURL)).default;
+    const isTS = isFileTS(filePath);
+
+    if (isTS) {
+        await import("tsx");
+    }
+
+    const config = (await import(fileURL.href)).default?.default ?? (await import(fileURL.href)).default;
```

Comment:
Why would this change be necessary? In which cases `(await import(fileURL)).default` wouldn't work?

Reply:
It looks like this line https://github.com/eslint/eslint/blob/0cb4914ef93cd572ba368d390b1cf0b93f578a9d/lib/eslint/eslint.js#L300 is causing `eslint.config.mts` files to have nested default exports (`default: { default: actualConfigArray }`), if we get rid of that line everything works fine with just `(await import(fileURL)).default`.